xl: Avoid multi-disks node to exit when one disk fails

vadmeste opened the pull request:
## Description
It makes sense that a node which has multiple disks to start when one
disk fails, returning i/o error for example. This commit will make this
faulty tolerence available in this specific use case.

## Motivation and Context
The cluster should be started when a disk is broken.

## How to test this PR?
Contact me.


## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

harshavardhana reviewed (changes requested):
why do we need this? @vadmeste

Commit:
xl: Avoid multi-disks node to exit when one disk fails

It makes sense that a node which has multiple disks to start when one
disk fails, returning i/o error for example. This commit will make this
faulty tolerence available in this specific use case.

poornas commented on cmd/prepare-storage.go:
```diff
@@ -72,78 +69,37 @@ var printEndpointError = func() func(Endpoint, error, bool) {
 
-			// Renames and schedules for puring all bucket metacache.
-			renameAllBucketMetacache(epPath)
+	// Renames and schedules for puring all bucket metacache.
```
```suggestion
	// Renames and schedules for purging all bucket metacache.
```

poornas commented on cmd/tier-journal.go:
```diff
@@ -56,23 +56,29 @@ var (
+	for _, diskPath := range diskPaths {
+		j := &tierJournal{
+			diskPath: diskPath,
+		}
```
the tier deletion journal is intentionally being done centrally as it an overkill to have it run on all disk paths. @krisis PTAL

krisis replied:
@vadmeste to select the first available local disk on a server works to an extent. It's definitely an improvement. Additionally, we need to act on journals that may be present on other local disks, due to a different set of disks available locally during previous server restarts.

E.g 
1. On the first startup, when all disks are online, this approach would select the first local disk on each server.
2. Imagine the first disk on the first server goes temporarily offline. On restart, we would select the second disk on this server.
3. Now the first disk is back online. On a subsequent restart, we would pick this disk. 

At this point we would have a journal in the second disk which will remain unattended. We need to address this too.

krisis replied:
@vadmeste we could address the issue of possibly unattended tier deletion journals on other local disks in a separate PR.

vadmeste replied:
Yes, but I just thought this cannot be more bad than what we have right now. Yes, we can revert this and address it more properly in another PR.

poornas reviewed (approved):
LGTM

Commit:
journal tries the next local disk when error
